ibis/common/tests/test_grounds.py

```diff
@@ -192,6 +192,28 @@ class NoHooves(Farm):
     assert g1.goats == 0
 
 
+def test_keyword_argument_reordering():
+    class Alpha(Annotable):
+        a = IsInt
+        b = IsInt
+
+    class Beta(Alpha):
+        c = IsInt
+        d = Optional(IsInt, default=0)
+        e = IsInt
```

Comment:
Ah, so the declaration can be in any order, but ultimately the signature has to follow the Python semantics.